cmd/snap/cmd_run.go

```diff
@@ -702,6 +706,181 @@ func (x *cmdRun) runCmdUnderGdb(origCmd, env []string) error {
 	return gcmd.Run()
 }
 
+type SnapTrace struct {
+	TotalTime    float64
+	execRuntimes []ExecRuntime
+}
+
+func (st *SnapTrace) ExecRuntimes() []ExecRuntime {
+	return st.execRuntimes
+}
+
+type perfStart struct {
+	Start  float64
+	Execve string
+}
+
+type ExecRuntime struct {
+	Execve   string
+	TotalSec float64
+}
+
+type byRuntimeAsc []ExecRuntime
+
+func (a byRuntimeAsc) Len() int           { return len(a) }
+func (a byRuntimeAsc) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byRuntimeAsc) Less(i, j int) bool { return a[i].TotalSec < a[j].TotalSec }
```

Comment:
Wonder if we should make it descending, so that the longest executions come first.

Reply:
I played with this a bit and removed the sorting again in f944c3d - maybe a bit too much YAGNI but I was thinking that in a sample of ~10 execs its easy enough to see what is slow and it might help users when they see the execs in the same order as they appear in the wrapper/desktop-launcher?